localstack/services/dynamodb/provider.py

```diff
@@ -356,15 +359,120 @@ def modify_context_region(context: RequestContext, region: str):
         context.request.headers["Authorization"] = original_authorization
 
 
+class DynamoDBDeveloperEndpoints:
+    """
+    Developer endpoints for DynamoDB
+    DELETE /_aws/dynamodb/expired - delete expired items from tables with TTL enabled; return the number of expired
+        items deleted
+    """
+
+    @route("/_aws/dynamodb/expired", methods=["DELETE"])
+    def delete_expired_messages(self, _: Request):
+        no_expired_items = delete_expired_items()
+        return {"ExpiredItems": no_expired_items}
+
+
+def delete_expired_items() -> int:
+    """
+    This utility function iterates over all stores, looks for tables with TTL enabled,
+    scan such tables and delete expired items.
+    """
+    no_expired_items = 0
+    for account_id, region_name, state in dynamodb_stores.iter_stores():
+        ttl_specs = state.ttl_specifications
+        client = connect_to(region_name=region_name).dynamodb
```

Comment:
```suggestion
        client = connect_to(aws_access_key_id=account_id, region_name=region_name).dynamodb
```

Otherwise the expired items will never be cleared in tables belonging to non-default accounts.